src/Tasks/Microsoft.NET.Build.Tasks/PrepareForReadyToRunCompilation.cs

```diff
@@ -221,32 +232,47 @@ bool InputFileEligibleForCompilation(ITaskItem file)
                     using (var pereader = new PEReader(fs))
                     {
                         if (!pereader.HasMetadata)
-                            return false;
-
-                        if ((pereader.PEHeaders.CorHeader.Flags & CorFlags.ILOnly) != CorFlags.ILOnly)
-                            return false;
+                        {
+                            return Eligibility.None;
+                        }
 
                         MetadataReader mdReader = pereader.GetMetadataReader();
                         if (!mdReader.IsAssembly)
                         {
-                            return false;
+                            return Eligibility.None;
+                        }
+
+                        if ((pereader.PEHeaders.CorHeader.Flags & CorFlags.ILOnly) != CorFlags.ILOnly)
+                        {
+                            return Eligibility.ReferenceOnly;
                         }
 
                         // Skip reference assemblies and assemblies that reference winmds
                         if (ReferencesWinMD(mdReader) || IsReferenceAssembly(mdReader) || !HasILCode(pereader, mdReader))
```

Comment:
We should probably return `Eligibility.None` when `IsReferenceAssembly` returns true to prevent crossgen from referencing non-implementation assemblies.

Reply:
OK, this should not happen, but it cannot hurt to be defensive and checking Eligibility at all for references is defensive. The list coming in should be all managed implementation assemblies already.